src/App.tsx

```diff
@@ -1,54 +1,135 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import './App.scss';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  const filteredPeople = useMemo(() => {
+    if (!query && isInputFocused) {
+      return peopleFromServer;
+    }
+
+    return peopleFromServer.filter(person => person.name.toLowerCase()
+      .includes(query.toLowerCase()));
+  }, [query, isInputFocused]);
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleBlur = useCallback(
+    () => {
+      setTimeout(() => {
+        setIsDropdownVisible(false);
+        setIsInputFocused(false);
+      }, 500);
+    }, [],
+  );
+
+  const handleFocus = useCallback(
+    () => {
+      setIsInputFocused(true);
+      setIsDropdownVisible(true);
+    }, [],
+  );
+
+  const handleSelect = useCallback(
+    (
+      event: React.MouseEvent<HTMLAnchorElement>,
+      person: Person,
+    ) => {
+      event.preventDefault();
+      setSelectedPerson(person);
+      setQuery(person.name);
+      setIsDropdownVisible(false);
+    }, [],
+  );
+
+  const handleReset = () => {
+    setQuery('');
+    setSelectedPerson(null);
+  };
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setIsDropdownVisible(!query || isInputFocused);
+    }, 500);
+
+    return () => clearTimeout(timerId);
+  }, [query, isInputFocused, handleFocus, handleBlur]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
-        <div className="dropdown-trigger">
+      <div className={`dropdown ${isDropdownVisible ? 'is-active' : ''}`}>
+        <div className="dropdown-trigger has-icons-right">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleInput}
+            onBlur={handleBlur}
+            onFocus={handleFocus}
           />
+          {!!query && (
+            <span className="icon is right">
+              <button
+                className="delete is small"
+                type="button"
+                onClick={handleReset}
+              >
+                x
+              </button>
+            </span>
```

Comment:
put this button inside input please

it should be like that
![image](https://github.com/mate-academy/react_autocomplete/assets/91946036/a28bf431-62ce-4ebe-ae4d-d70c993fb6a8)